### ISSUE TITLE
Using custom latex style file effortlessely; option to turn off all non-essential sphinx latex formatting 

### ISSUE THREAD

**Issue by austinbenny** [sphinx-doc/sphinx#11717]
**Is your feature request related to a problem? Please describe.**
I am trying to use my custom latex style file "effortlessly" with sphinx. This feature request is born from this need.

**Describe the solution you'd like**
A flag in conf.py that turns off most sphinx formatting and/or an option to include my custom style file and have it override all sphinx formatting.

**Describe alternatives you've considered**
I have explored `latex_additional_files` and `latex_elements` to no avail.

**Additional context**
In my workflow, the PDF output is more important than the HTML output. I have read the docs section [here](https://www.sphinx-doc.org/en/master/latex.html) that outlines latex customization but wanted to ask this general question to see what others did.

I have a latex style file that i use for my current template and am looking for a way to "plug-n-play" this style file with sphinx. The latex style is pretty complicated so ideally, I can turn off all sphinx latex style and class files, and just use mine. I'm aware this might not be possible but am wondering what my options are?

I looked at the latex_additional_files argument and it looks promising.

TL;DR: i have a highly customized latex style file that i would like to use effortlessly with sphinx.

- [e.g. URL or Ticket]

**Comment by austinbenny** [comment]
Edit: i'm seeing another option in the docs that could be helpful `latex_theme_path`. But documentation is sparse. I can elaborate on my problem if this option is worth looking into.

**Comment by picnixz** [comment]
I only sparsely worked with the latex backend so I cannot say for sure that we could extract a minimal set of macros for rendering our documents. 

There are actually a lot of hard coded macros that could be customised but this requires going through the code base and locate them, in addition to document them.

Which aspect of LaTeX do you want to customise?

**Comment by mgeier** [comment]
I have wanted to do a similar thing and came up with a work-around that might be useful: https://github.com/spatialaudio/nbsphinx/tree/master/include-in-latex

I used this approach to include two Sphinx projects as appendices in an otherwise hand-written Sphinx document: https://github.com/mgeier/diss ([PDF](https://github.com/mgeier/diss/releases/download/initial-submission/Geier_Diss.pdf)).

**Comment by austinbenny** [comment]
@picnixz , if i change my `conf.py` like this, i turn off most of the sphinx formatting.

```python
latex_additional_files = ["logo.pdf", "customStyle.sty"]
latex_theme = "howto"
latex_elements = {
    'classoptions' : 'oneside',
    'preamble': r"\usepackage{customStyle}",
    "fontpkg": "",
    "fncychap": "",
    "geometry": "",
    "maketitle": "",
    "releasename": "",
    "tableofcontents": "",
    "transition": "",
    "makeindex": "",
    "printindex": "",
    "fvset": "",
    }
```

I could probably optimize the dict above to remove anything that is off by default (instead of being explicit). Here are the remaining formatting items i want to turn off or edit:

* The sphinx caption on the figures and tables is "Fig. 1" and "Table 1" respectively. My custom template has a different text/caption: "Figure 4-1" and "Table 4-1" where 4 is the section number. I have code in my custom file to change it, but it looks like in `sphinxmessages.sty`, the caption styles are hard-coded?
* The color for all the titles is blue. I think i can figure out how to make it all black in my custom style file, but it would be good to revert to the latex default (black) if i could toggle a key somewhere in the `conf.py` and not worry about adding to my custom style file.
* I have to find a way to reconcile how sphinx uses references and glossaries versus how my style file does, but i think i will differ to sphinx here and turn off the references and glossary code in my style file.

I cant see anything else yet, but it looks like those changes to my `conf.py` got me halfway there.

**Comment by austinbenny** [comment]
@mgeier 
Hmm, this is helpful. I need to dig in to extract the things i need, but essentially my problem is i want to make sphinx play better with latex, but it looks like you made latex play better with sphinx -- you solved the inverse of my problem haha.

**Comment by austinbenny** [comment]
@mgeier @picnixz , there is an option for `latex_theme_path` and `latex_theme`. I'm thinking if i can turn my style file into a class file, and pass it in here, i can see some sort of result.

**Comment by apthorpe** [comment]
I have the same issue - for formal QA, documents must be in a very particular format specific to my organization. In the past I have used Doxygen and heavily modified its default document style to customize the title page, headers, footers, etc. I attempted to use a `latex_document` tuple and `latex_additional_files` to import a renamed version of `sphinxmanual.cls` (changing only the file name and arguments to `\ProvidesClass`) but something is not getting set correctly. I've had no success with `latex_docclass` (documentation doesn't really explain how to use it).

Most of the modifications I need are small changes to the default documentclass (onepage, all arabic page numbers, etc.) and the only path forward seems to be to manually replace `_build/latex/sphinxmanual.cls` between running Sphinx and rendering with LaTeX.
